apache/incubator-metron
# METRON-385 Create Ambari Service Definition for Indexing

https://issues.apache.org/jira/browse/METRON-385

Opening this PR for community review and feedback.

Prereq:
* Ambari 2.4

Testing this PR:
* The RPMs in metron-deployment should be built.  They should either be mounted or copied to /localrepo in the node to be used as the indexing master node.
* Copy or mount the services and stack contents into the appropriate places on the ambari-server node.  Afterwards, restart the Ambari server if it is running: ```service ambari-server restart```. For example:
  * incubator-metron/metron-deployment/packaging/ambari/src/main/resources/stacks/HDP/2.4/services/INDEXING -> /var/lib/ambari-server/resources/stacks/HDP/2.4/services/INDEXING
  * incubator-metron/metron-deployment/packaging/ambari/src/main/resources/common-services/INDEXING -> /var/lib/ambari-server/resources/common-services/INDEXING
* Install the master to a node with the Kafka Broker.

Limitations / Notes
* Service advisor for ensuring that the service is colocated appropriately doesn't appear to work. It's currently in the PR as-is.  It can easily be removed if we want to leave it out for now.
* Because the RPMs are currently not published to a yum repo, it's necessary to do the copy / mount step. Assuming the RPMs find a more permanent home, the repolist for the used stack should be updated and preferred.
* Elasticsearch.properties had a couple properties added to reflect what was in the Ansible scripts, because the current file was missing properties necessary to run the script outside of that context.
* Ambari currently provides no hooks for performing actions when a service is deleted.  To ensure cleanup please run

```
#!/bin/bash

# kafka
/usr/hdp/current/kafka-broker/bin/kafka-topics.sh \
        --zookeeper $ZK_QUORUM \
        --delete \
        --topic indexing
done

# storm kill
storm kill indexing

# yum delete
yum -y erase metron-indexing metron-common

# final file delete
rm -rf /usr/metron
```
